### ISSUE TITLE
2019-12-03 tc017 M4SD AEPCT values have the wrong units

### ISSUE THREAD

**Issue by tensfeldt** [tensfeldt/openNCA#79]
When running tc017 unchanged the values for AEPCT have the wrong units as shown in the output from `data_out` below.

```
      AMT.4.00 AMT.12.00 AMT.24.00    AE.4.00    AE.12.00  AE.24.00 AEPCT.4.00 AEPCT.12.00 AEPCT.24.00       AE    AEPCT
1  25.05333333 41.907250 13.426541  7.2444471 25.05333333 66.960583   80387125    87631572   8351111.1 22.32019 26795708
2  13.62846961 12.426464  6.575190  5.4580767 13.62846961 26.054933   32630123    38088200  13628469.6 26.05493 32630123
3  12.21048529 16.376961  5.168075  3.5434475 12.21048529 28.587446   33755521    37298969  12210485.3 28.58745 33755521
4  27.72319608 51.963229 17.789647 13.9923770 27.72319608 79.686425   97476073   111468450   9241065.4 26.56214 32492024
5   0.02410104  3.166148  1.897270  1.0167127  0.02410104  3.190249    5087519     6104232     80336.8 10.63416 16958398
6  24.69017647 23.201221 11.575106 11.2366114 24.69017647 47.891397   59466503    70703114  12345088.2 23.94570 29733251
7  23.09682255 55.688115 17.827652 16.1241618 23.09682255 78.784937   96612589   112736751   7698940.8 26.26165 32204196
8  30.69431373 24.169067  9.693353  8.3827214 30.69431373 54.863380   64556733    72939455  15347156.9 27.43169 32278367
9   6.30994510 20.314902  6.153111  3.5060749  6.30994510 26.624847   32777958    36284033   6309945.1 26.62485 32777958
10 30.69894902 31.441829 13.062466 10.5992059 30.69894902 62.140778   75203244    85802450  15349474.5 31.07039 37601622
11  3.14810706  4.077009  2.046223  1.6837875  3.14810706  7.225116    9271339    10955127  10493690.2 24.08372 30904464
12  1.94718882  4.163677  1.862071  1.1313144  1.94718882  6.110866    7972938     9104252   6490629.4 20.36955 26576458
13  4.37975000 20.245592  5.484842  4.3758682  4.37975000 24.625342   30110184    34486052   4379750.0 24.62534 30110184
14  1.62334324 56.971298  8.057586  4.1501386  1.62334324 58.594641   66652228    70802366    811671.6 29.29732 33326114
15 41.92012941 35.572466 11.807324 18.0401056 41.92012941 77.492595   89299919   107340024  13973376.5 25.83087 29766640
16  2.03049647  5.085595  1.261559  1.3372635  2.03049647  7.116091    8377650     9714914   6768321.6 23.72030 27925502
17 15.39428627 28.948612 13.206786  5.2331135 15.39428627 44.342898   57549684    62782798   5131428.8 14.78097 19183228
18 24.76771176 24.630000  8.869531  6.5233800 24.76771176 49.397712   58267243    64790623  12383855.9 24.69886 29133622
19  7.53830000 22.008810  4.816471  4.9030369  7.53830000 29.547110   34363580    39266617   7538300.0 29.54711 34363580
20 13.81669539 35.520000 11.195553 14.7010898 13.81669539 49.336695   60532248    75233338   6908347.7 24.66835 30266124
21  0.61596422 17.210330  6.034000  3.3852888  0.61596422 17.826295   23860295    27245583    615964.2 17.82629 23860295
22  1.39060441  3.776407  1.510945  0.9267487  1.39060441  5.167011    6677956     7604705   4635348.0 17.22337 22259854
23  0.55042791  4.409276  2.056919  1.7118627  0.55042791  4.959704    7016623     8728486   1834759.7 16.53235 23388744
24 46.25482647 37.172404 13.587868  9.2833358 46.25482647 83.427230   97015098   106298434  15418275.5 27.80908 32338366
```

**Comment by jhhughes256** [comment]
### Update

When running tc008 M4SS I noticed that when using `debug()` on `aepct()` that `AE` was in ng while `DOSE` was in mg. 

```
> debug(aepct)
> results_list <- run_computation(data = d, map = mct, flag = flags, 
  parameterset = parameterset)
Browse[2]> ae
[1] 12738576
Browse[2]> dose
[1] 100
Browse[2]> Q
> undebug(aepct)
```

The unit conversion output from this model when not run with aepct on debug is included below.

```
unit_conversion : Unit Class 1 (Time) time_col:  TMAXRATE MIDPTLAST TLAG KELTMLO KELTMHI THALF MIDPT1  parameters are scaled from  HR  to  HR  via scaling factor:  1 
unit_conversion : Unit Class 2 (Amount) amount_col:  AMT.24.00 AE.24.00 AE AETAU1  parameters are scaled from  NG  to  MG  via scaling factor:  1e-06 
unit_conversion : Unit Class 3 (Dose) dose_col:  DOSE1  parameters are scaled from  MG  to  MG  via scaling factor:  1 
unit_conversion : Unit Class 4 (Volume) volume_col:  VOLSUM  parameters are scaled from  ML  to  L  via scaling factor:  0.001 
unit_conversion : Unit Class 6 (1/Time) kel_col:  KEL  parameters are scaled from  1/HR  to  1/HR  via scaling factor:  1 
unit_conversion : Unit Class 11: ([Volume.Amount]/Volume) aurc_col:  AURCALL AURCLAST AURC2_18 AURC8_16 AURCINFO AURCINFP  parameters are scaled from  ML.NG/ML  to  ML.UG/ML  via scaling factor:  0.001 
unit_conversion : Unit Class 13: (Amount/Time) rate_col:  MAXRATE RATELAST RATE1  parameters are scaled from  NG/HR  to  UG/HR  via scaling factor:  0.001 
```

**Comment by tensfeldt** [comment]
This issue is still occurring with commit be5ef37

**Comment by jhhughes256** [comment]
This issue still occurs as of 5a9f91aa0d.

```r
> r
       SDEID DOSE    AMT.4.00 AMT.12.00 AMT.24.00   AE.4.00  AE.12.00   AE.24.00 AEPCT.4.00 AEPCT.12.00 AEPCT.24.00         AE       AEPCT
1   58098661  200  1.62334324 56.971298  8.057586 58.594641 66.652228  70.802366   29297321    33326114    35401183  70.802366 35401183088
2   58613088  300 41.92012941 35.572466 11.807324 77.492595 89.299919 107.340024   25830865    29766640    35780008 107.340024 35780008072
3  107790547  300 25.05333333 41.907250 13.426541 66.960583 80.387125  87.631572   22320194    26795708    29210524  87.631572 29210523969
4  109463837  100 13.62846961 12.426464  6.575190 26.054933 32.630123  38.088200   26054933    32630123    38088200  38.088200 38088199902
5  114635196  100 12.21048529 16.376961  5.168075 28.587446 33.755521  37.298969   28587446    33755521    37298969  37.298969 37298968529
6  190133875  300 27.72319608 51.963229 17.789647 79.686425 97.476073 111.468450   26562142    32492024    37156150 111.468450 37156149843
7  191063622   30  0.02410104  3.166148  1.897270  3.190249  5.087519   6.104232   10634164    16958398    20347441   6.104232 20347440595
8  206689591  200 24.69017647 23.201221 11.575106 47.891397 59.466503  70.703114   23945699    29733251    35351557  70.703114 35351557157
9  223523507  300 23.09682255 55.688115 17.827652 78.784937 96.612589 112.736751   26261646    32204196    37578917 112.736751 37578916993
10 246293231  200 30.69431373 24.169067  9.693353 54.863380 64.556733  72.939455   27431690    32278367    36469727  72.939455 36469727353
```

**Comment by rudraya** [comment]
This issue was addressed in b9591d0, please let me know if it still occurs.

**Comment by tensfeldt** [comment]
Confirmed that commit b9591d0 resolves the issue with tc017 M4SD and units for AEPCT. Checked AMT/AET/AEPCTT/AE/AEPCT/RATE and all values for these match manually calculated parameters.
This issue can be closed.
